File: kotlin-junit/src/test/kotlin/varar/yahtzee.steps.kt
@file:JvmName("YahtzeeSteps")

package varar

import dev.varar.kotlin.sensor
import dev.varar.kotlin.steps
import examples.score

val yahtzeeSteps = steps {
    sensor("Examples of dice, category and score") { row: Map<String, String> ->
        val dice = row.getValue("dice").split(",").map { it.trim().toInt() }
        mapOf("score" to score(dice, row.getValue("category")))
    }
}

File: rust-cargotest/src/varar/yahtzee.steps.rs
use super::Ctx;
use crate::yahtzee::score;
use ::varar::Steps;
use std::collections::BTreeMap;

pub fn register(s: &mut Steps<Ctx>) {
    // Header-bound row: the row arrives keyed by column, and the computed
    // columns go back the same way for the core to diff cell by cell.
    s.sensor("Examples of dice, category and score", |_ctx: Ctx, row: BTreeMap<String, String>| {
        let dice: Vec<i64> = row["dice"]
            .split(',')
            .map(|d| d.trim().parse().expect("die"))
            .collect();
        Ok(BTreeMap::from([("score".to_string(), score(&dice, &row["category"]).to_string())]))
    });
}
